### PR TITLE
Implement borrowing list & detail endpoint

### PR DESCRIPTION
**Pull request by Oxbay**
It`s a draft version, so everything can change)

### CODE REVIEW

File: borrowing/models.py [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+
 
 class Borrowing(models.Model):
-    pass
+    borrow_date = models.DateField()
+    expected_return_date = models.DateField()
+    actual_return_date = models.DateField(null=True, blank=True)
+    book_id = models.ForeignKey(Book, on_delete=models.CASCADE)
+    user_id = models.ForeignKey("User", on_delete=models.CASCADE)
+
```
Comment (Familenko) [inline borrowing/models.py:14]: better put here 
```
user = models.ForeignKey(get_user_model(), on_delete=models.CASCADE)
```

and put related_name

File: borrowing/urls.py [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
-urlpatterns = [
+from borrowing.views import (
+    BorrowingSerializer,
+    BorrowingListView,
+    BorrowingDetailView,
+)
 
+urlpatterns = [
+    path("borrowings/", BorrowingListView.as_view(), name="borrowing-list"),
+    path(
```
Comment (Familenko) [inline borrowing/urls.py:11]: please keep "borrowings/" for POST method

**Review by Familenko — CHANGES_REQUESTED** [review]

File: borrowing/urls.py [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
 
-urlpatterns = [
+from borrowing.views import (
+    BorrowingSerializer,
+    BorrowingListView,
+    BorrowingDetailView,
+)
 
+urlpatterns = [
+    path("borrowings_list/", BorrowingListView.as_view(), name="borrowing-list"),
```
Comment (Sel-Fisher) [inline borrowing/urls.py:10]: better just borrowings with out list?

File: borrowing/views.py [hunk 4]
```diff
@@ -10,5 +8,19 @@
...
+class BorrowingDetailView(generics.RetrieveAPIView):
+    queryset = Borrowing.objects.select_related("book", "user")
+    serializer_class = BorrowingSerializer
+
+
+class UserBorrowingsListView(generics.ListAPIView):
+    serializer_class = BorrowingSerializer
+
+    def get_queryset(self):
+        user = self.request.user
```
Comment (Sel-Fisher) [inline borrowing/views.py:25]: use self.request.user.pk to have same results but avoid problem when unathenticated user try to visit this page

**Review by Sel-Fisher — APPROVED** [review]
Very cool!

File: borrowing/urls.py (hunk 3 above)
Comment (Familenko) [reply borrowing/urls.py:10]: I told him to do this according documentation

**Review by Familenko — APPROVED** [review]

**Pull request merged**